OdenTech/sse
# fixing memory leak when creating a stream that already exists

## Bug Description
When creating a stream via `CreateStream()`, a new stream is always created even if a stream with the same `id` already exists, causing a memory leak where goroutines are created via `str.run()` but never closed. 

## How To Read This PR
This PR fixes that bug by closing the newly created stream if the stream `id` is found in the server's map of streams.

## How to Test This PR
These changes are reflected in the tests, but there is a potential race condition in the test where the newly created stream may not have been deleted by the time we check whether `runtime.NumGoroutine()` is equal to `numGoRoutines`. 

At the moment I can't find a better, simple way to test this so if you'd like me to remove the test let me know. 

```
make test
```
